src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -278,6 +278,18 @@ impl<'a> Linker for GccLinker<'a> {
     }
 
     fn set_output_kind(&mut self, output_kind: LinkOutputKind, out_filename: &Path) {
+        if self.sess.target.target.target_os == "vxworks"
+            && matches!(
+                output_kind,
+                LinkOutputKind::StaticNoPicExe
+                    | LinkOutputKind::StaticPicExe
+                    | LinkOutputKind::StaticDylib
+            )
+        {
+            // FIXME: What this option does? Is it necessary given that we are already passing
+            // `-static` and `-nodefaultlibs`? VxWorks GNU toolchain has no public documentation.
+            self.cmd.arg("--static-crt");
```

Comment:
@BaoshanPang, could you comment on this?

Reply:
For VxWorks, we use "--static-crt" to tell the compiler driver that we want to use static C libraries no matter which kind of rust libraries will be used.

Reply:
@BaoshanPang 
That's why I was mentioning `-static` and `-nodefaultlibs`, which `rustc` passes to gcc-like linkers, in case of VxWorks as well.
How does `wr-c++` interpret those options?

If it's gcc-compatible, I'd expect `-nodefaultlibs` to remove libc from linking altogether and `-static` to provide the desired "link everything statically" semantic.

Reply:
@BaoshanPang 
One more question, since you added VxWorks support to https://github.com/rust-lang/libc as well.
Do I understand correctly that on VxWorks libc is linked into Rust programs due to this fallback clause?
https://github.com/rust-lang/libc/blob/master/src/unix/mod.rs#L351-L355

Reply:
Wait a second, was `--static-crt` introduced to the `wr-c++` driver for `rustc` specifically by any chance?
To workaround the default behavior of `#[link(name = "c")]` linking libc dynamically?

Reply:
> Wait a second, was `--static-crt` introduced to the `wr-c++` driver for `rustc` specifically by any chance?
> To workaround the default behavior of `#[link(name = "c")]` linking libc dynamically?

Yes.

Reply:
> @BaoshanPang
> One more question, since you added VxWorks support to https://github.com/rust-lang/libc as well.
> Do I understand correctly that on VxWorks libc is linked into Rust programs due to this fallback clause?
> https://github.com/rust-lang/libc/blob/master/src/unix/mod.rs#L351-L355

No, I was not aware of this part when I added the code for rust libc.

Reply:
> `-static` to provide the desired "link everything statically" semantic.

One of the combination we want to support is shared rust library with static VxWorks C library.

Reply:
@BaoshanPang 
I see, thanks for the information!
I'll keep the current logic as is then, since I cannot test the changes myself.

However, the recommended way to link to libc, libpthread and similar system libraries is still through `#[link]` attributes in https://github.com/rust-lang/libc, with `cfg(target_feature = "crt-static")` being used for switching between dynamic and static linking.
See https://github.com/rust-lang/libc/blob/master/src/unix/mod.rs#L295-L357 for examples from other targets.

Reply:
> @BaoshanPang
> I see, thanks for the information!
> I'll keep the current logic as is then, since I cannot test the changes myself.
> 
> However, the recommended way to link to libc, libpthread and similar system libraries is still through `#[link]` attributes in https://github.com/rust-lang/libc, with `cfg(target_feature = "crt-static")` being used for switching between dynamic and static linking.
> See https://github.com/rust-lang/libc/blob/master/src/unix/mod.rs#L295-L357 for examples from other targets.

Thanks for the information, We may use this solution for the next release.